datajunction-clients/python/datajunction/_internal.py

```diff
@@ -571,6 +574,22 @@ def _create_tag(
         )
         return response
 
+    def _list_nodes_with_tag(
+        self,
+        tag_name: str,
+        node_type: Optional[models.NodeType] = None,
+    ):
+        """
+        Retrieves all nodes with a given tag.
+        """
+        response = self._session.get(
+            f"/tags/{tag_name}/nodes"
+            + (f"?node_type={node_type.value}" if node_type else ""),
+        )
+        if response.status_code == 404:
```

Comment:
If the response is a 404, does that typically mean the tag doesn't exist? If that's the case is it better here to inform the user that the tag doesn't exist rather than defaulting to an empty list of nodes?

Reply:
Hmm... I think you may be right. Will fix.